server/src/test/java/org/elasticsearch/cluster/routing/allocation/AwarenessAllocationTests.java

```diff
@@ -496,6 +501,7 @@ public void testMoveShardOnceNewNodeWithAttributeAdded6() {
 
         logger.info("--> complete relocation");
         clusterState = startInitializingShardsAndReroute(strategy, clusterState);
+        clusterState = startInitializingShardsAndReroute(strategy, clusterState);
```

Comment:
This doesn't look good. I checked this code out locally and this change is necessary to keep tests from working. I don't think we should break the logic for existing cases for them to require two iterations of the decider to arrive at the correct allocation?

Reply:
@original-brownbear  Yes，Thank you for your good advice. You made a lot of sense. I've changed it.